--- a/ops/kda/chunk_intra_token_parallel.py
+++ b/ops/kda/chunk_intra_token_parallel.py
@@ -43,6 +43,7 @@ def chunk_kda_fwd_kernel_intra_token_parallel(
     T,
     H: tl.constexpr,
     K: tl.constexpr,
+    BK: tl.constexpr,
     BT: tl.constexpr,
     BC: tl.constexpr,
     BH: tl.constexpr,
@@ -89,7 +90,6 @@ def chunk_kda_fwd_kernel_intra_token_parallel(
     Akk += bos * H*BC
     beta += bos * H
 
-    BK: tl.constexpr = triton.next_power_of_2(K)
     o_h = tl.arange(0, BH)
     o_k = tl.arange(0, BK)
     m_h = (i_hg * BH + o_h) < H
@@ -146,6 +146,7 @@ def chunk_kda_fwd_intra_token_parallel(
     N = len(cu_seqlens) - 1 if cu_seqlens is not None else B
     BT = chunk_size
     BC = sub_chunk_size
+    BK = triton.next_power_of_2(K)
 
     def grid(meta): return (B * T, triton.cdiv(H, meta['BH']))
     chunk_kda_fwd_kernel_intra_token_parallel[grid](
@@ -162,6 +163,7 @@ def chunk_kda_fwd_intra_token_parallel(
         T=T,
         H=H,
         K=K,
+        BK=BK,
         BT=BT,
         BC=BC,
     )
